app/Http/Controllers/Plan/PlansController.php

```diff
@@ -609,40 +612,59 @@ public function completeDay(Request $request, $day_id)
         $user_is_member = $this->compareProjects($user->id, $this->key);
 
         if (!$user_is_member) {
-            return $this->setStatusCode(401)->replyWithError(trans('api.projects_users_not_connected'));
+            return $this
+                ->setStatusCode(SymfonyResponse::HTTP_UNAUTHORIZED)
+                ->replyWithError(trans('api.projects_users_not_connected'));
         }
 
         $plan_day = PlanDay::where('id', $day_id)->first();
 
         if (!$plan_day) {
-            return $this->setStatusCode(404)->replyWithError('Plan Day Not Found');
+            return $this
+                ->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)
+                ->replyWithError('Plan Day Not Found');
         }
 
-        $user_plan = UserPlan::join('plans', function ($join) use ($user) {
-            $join->on('user_plans.plan_id', '=', 'plans.id')->where('user_plans.user_id', $user->id);
-        })->where('user_plans.plan_id', $plan_day->plan_id)
-            ->select('user_plans.*')
-            ->first();
+        $user_plan = UserPlan::getByPlanIdAndUserId($plan_day->plan_id, $user->id);
 
         if (!$user_plan) {
-            return $this->setStatusCode(404)->replyWithError('User Plan Not Found');
+            return $this
+                ->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)
+                ->replyWithError('User Plan Not Found');
         }
 
         $complete = checkParam('complete') ?? true;
         $complete = $complete && $complete !== 'false';
 
-        $user_plan = \DB::transaction(function () use ($complete, $plan_day, $user, $user_plan) {
-            if ($complete) {
-                $plan_day->complete($user->id);
-            } else {
-                $plan_day->unComplete($user->id);
+        $result = null;
+        try {
+            \DB::transaction(function () use ($complete, $plan_day, $user) {
+                if ($complete) {
+                    $plan_day->complete($user->id);
+                } else {
+                    $plan_day->unComplete($user->id);
+                }
+
+                return $plan_day;
+            });
+            $result = $complete ? 'completed' : 'not completed';
+        } catch (QueryException $e) {
+            // Catch only the error code for a duplicate entry
+            if ($e->getCode() == MySQLErrorCode::DUPLICATE_ENTRY) {
```

Comment:
log at info level and swallow the exception, return normal status code to user

Reply:
@bradflood I have updated it. The message will be logged at the 'info' level. If the exception is a "Integrity constraint violation: Duplicate entry", the endpoint will continue with the normal workflow and subsequently return a 200 code to user. see:

https://github.com/faithcomesbyhearing/dbp/pull/937/commits/296d9372d58d56d76d6f3881fdff2910acb12e39#diff-1a4ccaa9f272dd23f79692bdb76508f4605118b7c27469e381c4ca2edfa8f545R653